src/strategies/spidStrategy.ts

```diff
@@ -2,102 +2,78 @@
  * Builds and configure a Passport strategy to authenticate the proxy to the
  * different SPID IDPs.
  */
-
+// tslint:disable: no-object-mutation
+import { NonEmptyArray } from "fp-ts/lib/NonEmptyArray";
+import { NonEmptyString } from "italia-ts-commons/lib/strings";
 import * as SpidStrategy from "spid-passport";
+import { IDPEntityDescriptor } from "../types/IDPEntityDescriptor";
 import { SpidUser } from "../types/user";
+import { fetchIdpMetadata, parseIdpMetadata } from "../utils/idpLoader";
 import { log } from "../utils/logger";
 
-const spidStrategy = (
+/**
+ * Map provided idpMetadata in an object with idp key whitelisted in ipdIds.
+ * Mapping is based on entityID property
+ */
+export const mapIpdMetadata = (
+  idpMetadata: ReadonlyArray<IDPEntityDescriptor>,
+  idpIds: { [key: string]: string | undefined }
+) =>
+  idpMetadata.reduce(
+    (prev, idp) => {
+      const idpKey = idpIds[idp.entityID];
+      if (idpKey) {
+        prev[idpKey] = idp;
+        return prev;
+      }
+      log.warn(
+        `Unsupported SPID idp from metadata repository [${idp.entityID}]`
+      );
+      return prev;
+    },
+    {} as { [key: string]: IDPEntityDescriptor | undefined }
+  );
+
+const spidStrategy = async (
   samlKey: string,
   samlCallbackUrl: string,
   samlIssuer: string,
   samlAcceptedClockSkewMs: number,
   samlAttributeConsumingServiceIndex: number,
   spidAutologin: string,
-  spidTestEnvUrl: string
+  spidTestEnvUrl: string,
+  IDPMetadataUrl: string
+  // tslint:disable-next-line: parameters-max-number
 ) => {
-  const options = {
+  const IDP_IDS: { [key: string]: string | undefined } = {
+    "https://id.lepida.it/idp/shibboleth": "lepidaid",
+    "https://identity.infocert.it": "infocertid",
+    "https://identity.sieltecloud.it": "sielteid",
+    "https://idp.namirialtsp.com/idp": "namirialid",
+    "https://login.id.tim.it/affwebservices/public/saml2sso": "timid",
+    "https://loginspid.aruba.it": "arubaid",
+    "https://posteid.poste.it": "posteid",
+    "https://spid.intesa.it": "intesaid",
+    "https://spid.register.it": "spiditalia"
+  };
+
+  const idpMetadata = parseIdpMetadata(await fetchIdpMetadata(IDPMetadataUrl));
```

Comment:
we should split this operation (fetching + parsing) and add some logging, i.e. when the server starts I'd like to see:

```
Fetching SPID metadata from [...METADATA_URL...]...
Parsing SPID metadata...
Configuring IdPs...
```

Reply:
fixed in 0b9431d